enterprise_access/apps/content_assignments/models.py

```diff
@@ -0,0 +1,99 @@
+"""
+Models for content_assignments
+"""
+from django.db import models
+from django_extensions.db.models import TimeStampedModel
+from simple_history.models import HistoricalRecords
+
+from enterprise_access.apps.subsidy_access_policy.models import SubsidyAccessPolicy
+
+from .constants import LearnerContentAssignmentStateChoices
+
+
+class AssignmentPolicy(TimeStampedModel):
+    """
+    Manage the creation and lifecycle of LearnerContentAssignments according to configurable rules.
+    """
+    subsidy_access_policy = models.ForeignKey(
+        SubsidyAccessPolicy,
+        related_name="assignment_policy_config",
+        on_delete=models.CASCADE,
+        db_index=True,
+        unique=True,
+    )
+    # TODO: Add fields representing different configurable rules below.
+
+
+class LearnerContentAssignment(TimeStampedModel):
+    """
+    Represent an assignment of a piece of content to a learner.
+
+    .. pii: The learner_email field stores PII.
+    .. pii_types: email_address
+    .. pii_retirement: local_api
+    """
+    assignment_policy = models.ForeignKey(
+        AssignmentPolicy,
+        related_name="assignments",
+        on_delete=models.CASCADE,
+        db_index=True,
+        help_text="AssignmentPolicy defining the lifecycle rules of this assignment.",
+    )
+    learner_email = models.EmailField(
+        null=False,
+        blank=False,
+        unique=True,
```

Comment:
This can't be unique, a single email might be assigned multiple contents.

Reply:
We'll want to index this field, too.

Reply:
Doesn't have to be addressed in this PR, but we'll also eventually want a composite UNIQUE key on (email, content) or (email, content, state), depending on the approach we pick for dealing with the allocated -> errored -> (admin re-assigns) -> allocated lifecycle flow.

Reply:
whoops, carryover from the place I copied from